core/src/main/java/com/scottlogic/datahelix/generator/core/generation/string/generators/FakerGenerator.java

```diff
@@ -20,26 +20,25 @@
 import com.scottlogic.datahelix.generator.common.RandomNumberGenerator;
 import com.scottlogic.datahelix.generator.common.util.OrderedRandom;
 
-import java.util.function.Function;
 import java.util.stream.Stream;
 
 public class FakerGenerator implements StringGenerator {
 
     private final StringGenerator underlyingRegexGenerator;
-    private final Function<Faker, String> fakerFunction;
+    private final String fakerSpec;
     private final Faker randomFaker;
     private final Faker orderedFaker;
 
-    public FakerGenerator(StringGenerator underlyingRegexGenerator, Function<Faker, String> fakerFunction) {
+    public FakerGenerator(StringGenerator underlyingRegexGenerator, String fakerSpec) {
         this.underlyingRegexGenerator = underlyingRegexGenerator;
-        this.fakerFunction = fakerFunction;
+        this.fakerSpec = fakerSpec;
         randomFaker = new Faker();
         orderedFaker = new Faker(new OrderedRandom());
     }
 
     @Override
     public boolean matches(String string) {
-        return false;
+        return generateAllValues().anyMatch(fakerValue -> fakerValue.equals(string));
```

Comment:
I'm assuming `generateAllValues()` will stream values indefinitely - if so, this expression will hang.